### PR TITLE
misc: prepare 24.05 release

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

**Comment by MattSturgeon** [comment]
Will need to update some hard-coded branches in CI workflows too, especially the docs workflow.

**Comment by GaetanLepage** [comment]
> Will need to update some hard-coded branches in CI workflows too, especially the docs workflow.

The `build-doc.sh` script, yes. But this will be done in `main` after the branch-off.

**Review by MattSturgeon — APPROVED** [review]

File: docs/user-guide/install.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ You must use a `nixpkgs` version compatible with the nixvim version you choose.
 The `master` branch requires to use a _very recent_ version of nixpkgs unstable.
 In order to guarantee the compatibility between nixvim & nixpkgs it is recommended to always update both at the same time.
 
-When using a `stable` version you must use the corresponding nixvim branch, for example `nixos-23.11` when using NixOS 23.11.
+When using a `stable` version you must use the corresponding nixvim branch, for example `nixos-23.05` when using NixOS 23.05.
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply docs/user-guide/install.md:8]: Done in #1640

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply docs/user-guide/install.md:8]: And backported directly on the `nixos-24.05` branch in https://github.com/nix-community/nixvim/commit/2c25e77d8265a1b473a671d0cb5598c1830d94c6

**Pull request merged**